### ISSUE TITLE
notification_area: D-Bus context menu grabs mouse and keyboard forever when application is freezed

### ISSUE THREAD

**Issue by zaps166** [mate-desktop/mate-panel#1300]
#### Expected behaviour
X11 session is not freezed.

#### Actual behaviour
X11 session is freezed (invisible context menu grabs mouse and waits for freezed application to response).

#### Steps to reproduce the behaviour
1. Run application withh D-Bus tray protocol (e.g. Steam, Telegram, QMPlay2).
2. Application must freeze somehow, but easy way to reproduce is `killall app_process_name -SIGSTOP`.
3. Right-click on freezed application icon in notification area.

To unfreeze the X11 session, the only way is to switch to tty2, login, and kill the freezed application. Other solution used by many users is reset the computer.

#### MATE general version
1.26

#### Package version
1.26.2

#### Linux Distribution
Manjaro

---

The problem is in some D-Bus calls without reasonable timeout. I prefer to replace `-1` for about `1000` milliseconds in:
- `sn_dbus_menu_gen_call_event_sync`
- `sn_dbus_menu_gen_call_about_to_show_sync`

 I can do the PR - what do you think about this solution?

**Comment by lukefromdc** [comment]
I've had a number of general freezing issues presumed not related to MATE, when only the x11 session freezes I am lucky and
can change TTY out. Don't know if any of them were ever this bug however, as I've never had to kill Audacious or anything else in the
tray to get out of it, or maybe this was the case when I had to kill Xorg altogether. I am getting a lot of total video freezes
(sound files continue to play and it looks like video may continue to render though unsure of that, same for file transfers).
That happened with two different builds having in common only the first generation AMD GCN video card run with the
AMDGPU driver, the case, the drives, and a robust PSU that can handle plenty more power than needed.

I Cannot get another video card as I do not shop online nor with ID and Micro Center won't sell them for cash or without
ID as they are trying to limit volume purchases by cryptominers and resellers.  To get one I'd have to stand in the parking 
lot with cash, have someone else buy it with ID like a 14 year old buying beer, and hope I didn't get ripped off for a lot 
more than the price of cheap beer. Too much risk and cost over the video freezes, if the GPU dies I will have to switch to
an APU as those are not yet restricted.

Would be hard to diagnose any other freezes when the video freeze is the most common issue. They started after a round
of software updates in 2020 with my previous machine so something in the userspace Mesa driver pr X itself is the most likely 
culprit. The kernel was not changed at that time.

**Comment by muktupavels** [comment]
> The problem is in some D-Bus calls without reasonable timeout. I prefer to replace `-1` for about `1000` milliseconds in:
> 
>     * `sn_dbus_menu_gen_call_event_sync`
> 
>     * `sn_dbus_menu_gen_call_about_to_show_sync`

Are you sure these functions are responsible for freezes? These functions use `-1` as timeout which means default timeout... So function call will timeout after 25 seconds...

**Comment by zaps166** [comment]
Right, `-1` is a default timeout (25 seconds). It means 50 seconds of mouse and keyboard grab in overall, because there are 2 sync calls here: https://github.com/mate-desktop/mate-panel/blob/c4602959df4f2856e49ebf6b6c7a03be00d311f4/applets/notification_area/status-notifier/sn-dbus-menu.c#L265 It's too long and I never waited that long :sweat_smile:

I modified `sn_dbus_menu_gen_call_event_sync` and `sn_dbus_menu_gen_call_about_to_show_sync` to `1000` and it is quite ok for freeezed apps.

---

Imagine the application stops responding, you right-click on tray icon and your computer is locked for 50 seconds...

---

@lukefromdc It's not related to mouse and keyboard grab into a context menu window.

Try to look at `dmesg` to see `amdgpu` errors. I noticed sometimes older AMD GPUs runs less stable than Navi2.

> The kernel was not changed at that time.

Maybe it's time to upgrade, 5.15.x is stable for AMD GPUs. Maybe new Mesa crashes old kernel drivers? How about upgrading/downgrading `linux-firmware`?

**Comment by raveit65** [comment]
Confirmed with QMPlay2
